agent/grpc/public/services/peerstream/stream_tracker.go

```diff
@@ -142,6 +156,10 @@ type Status struct {
 	// - The error message when we failed to store a resource replicated FROM the peer.
 	// - The last error message when receiving from the stream.
 	LastReceiveErrorMessage string
+
+	// TODO(peering): consider keeping track of imported service counts thru raft
+	// ImportedServices is set that keeps track of which service names are imported for the peer
+	ImportedServices map[string]struct{}
```

Comment:
Why does this need to be a map instead of a simple int?

With an int, you could get rid of the `GetIMportedServicesCount` method. I would also renamed the other two functions to `IncrementImportedServiceCount` and `DecrementImportedServiceCount`

Reply:
IIUC, we were going to use a map for `ExportedServiceCount` and we wanted to use a map for consistency here, right? @freddygv (maybe I got this wrong)

per matt's comment above, I guess we could keep the imported services count an actual counter 🤔

Reply:
I cannot think of what else we would need to track for imported services.

For exported services, we likely need a map because we don't have the ability to check against memdb to see what services have been exported.

I guess a map is okay, but if we stick with it for imports then there is no reason to make the modifications to `handleUpdateService`. Instead you could add a `TrackImportedService` call to `handleUpsert` within the block handling service updates and a similar call to `RemoveTrackedService` within `handleDelete`. Both of those have the service name and with the map we wont double count services by tracking them before we know whether they actually are for a net-new service.